examples/nlp/sentence_embeddings_with_sbert.py

```diff
@@ -101,35 +159,61 @@
 indicates the least semantic similarity between the two sentences, and 5 indicates the
 most semantic similarity between the two sentences.
 
-The range of the cosine similarity is [-1, 1] and it's the output of the siamese network,
-but the range of the labels in the dataset is [0, 5]. We need to unify the range between
-the cosine similarity and the dataset labels, so while preparing the dataset, we will
-divide the labels by 2.5 and subtract 1.
+We will use `tensorflow_datasets` to load the data and `tf.data` API to preprocess the
+data.
+
+We will preprocess the sentences using `keras_nlp.models.RobertaPreprocessor` that will
+generate `token_ids`, and `padding_mask` which are the input for the RoBERTa model.
+
+The range of the cosine similarity scores is [-1, 1] and it's the output of the siamese
+network, but the range of the labels in the dataset is [0, 5]. We need to unify the range
+between the cosine similarity and the dataset labels, so while preparing the dataset, we 
+will divide the labels by 2.5 and subtract 1.
+
+We will use 2400 pairs for training data and 400 pairs for testing data.
 """
 
-TRAIN_BATCH_SIZE = 6
+TRAIN_BATCH_SIZE = 8
+TRAIN_NUM_BATCHS = 300
+
 VALIDATION_BATCH_SIZE = 8
+VALIDATION_NUM_BATCHS = 50
 
-TRAIN_NUM_BATCHS = 300
-VALIDATION_NUM_BATCHS = 40
+# Used to tune the value dynamically at runtime.
+AUTOTUNE = tf.data.AUTOTUNE
+
+preprocessor = keras_nlp.models.RobertaPreprocessor.from_preset("roberta_base_en")
 
-AUTOTUNE = tf.data.experimental.AUTOTUNE
+
+def one_dictionary(x):
+    """
+    Returns the data as one dictionary to be appropriate input for
+    `keras_core.Model`
+    """
+    return {
```

Comment:
Does keras_core support nested dictionaries?

Shouldn't this work:

{'input0': x[0], 'input1':x[1]}?